ui/server.go

```diff
@@ -50,13 +52,22 @@ func NewServer(port int) *Server {
 	}
 
 	return &Server{
-		srv: srv,
+		srv:       srv,
+		discovery: discovery.NewLANDiscoveryService(port),
 	}
 }
 
 // Serve starts the server
 func (s *Server) Serve() error {
 	log.Info(logPrefix, "server starting on: ", s.srv.Addr)
+
+	go func() {
+		err := s.discovery.Start()
+		if err != nil {
+			log.Error(logPrefix, "failed to start local discovery service: ", err)
```

Comment:
Should we proceed with starting web service if discovery fails?

Reply:
UI will work without it and discovery is optional. I think it enough just logs it here.